python-package/lightgbm/basic.py

```diff
@@ -974,6 +975,8 @@ def construct(self):
                         ctypes.c_int(used_indices.shape[0]),
                         c_str(params_str),
                         ctypes.byref(self.handle)))
+                    self.data = self.reference.data
+                    self.get_data()
```

Comment:
maybe we don't need the `get_data` here ?

Reply:
okay, I see. As only subset will call the slicing, I think the overhead cost is acceptable. 
Change the definition of `data` may cause many dependency problems.

Reply:
Yeah, it's expected that only subsetting branch of `construct` calls `get_data()`. Also, `get_data()` itself checks `used_indices`:
https://github.com/Microsoft/LightGBM/blob/1ff6c6d6bca05de2c13490264ba3b6bf7f993d4a/python-package/lightgbm/basic.py#L1392